optimum/exporters/onnx/model_patcher.py

```diff
@@ -600,8 +607,15 @@ def patched_forward(*args, **kwargs):
                 filtered_outputs[name] = outputs
 
             if is_transformers_version(">=", "4.48"):
-                if isinstance(filtered_outputs.get("past_key_values"), (DynamicCache, EncoderDecoderCache)):
-                    filtered_outputs["past_key_values"] = outputs["past_key_values"].to_legacy_cache()
+                if isinstance(filtered_outputs.get("past_key_values"), EncoderDecoderCache):
+                    if any("encoder" in output_name for output_name in config.outputs):
+                        filtered_outputs["past_key_values"] = filtered_outputs["past_key_values"].to_legacy_cache()
+                    else:
+                        filtered_outputs["past_key_values"] = filtered_outputs[
+                            "past_key_values"
+                        ].self_attention_cache.to_legacy_cache()
```

Comment:
is this needed since v4.56 or is it a fix that was needed since v4.48

Reply:
the need to specify this starts with 4.56, but it's also generally valid so I didn't see a need to condition it, wdyt ?
it seems that now the encoder decoder cache can contain output cross attention tensors even when they can't be reused (e.g. with bert-gpt2 encoder-decoder)